stdlib/marshal.gr

```diff
@@ -430,6 +434,9 @@ let marshal = (value, buf) => {
  * @param value: The value to serialize
  * @returns A byte-based representation of the value
  *
+ * @example Marshal.marshal(1) == b"\x03\x00\x00\x00"
+ * @example Marshal.marshal("🌾") == Marshal.marshal('🌾')
```

Comment:
I just tried this myself and it didn't seem to work. Is this a bug in our marshal implementation?

`Marshal.marshal("🌾") == Marshal.marshal("🌾")` also returned false when I tried it

Reply:
![image](https://github.com/grain-lang/grain/assets/40705786/6e6c6596-ecf7-4bc8-915b-f7f50a447e73)
I am thinking it may be an issue with how our byte comparison works just given the printed byte values

Reply:
The biggest problem with the example is that we're comparing the string `"🌾"` to the char `'🌾'`.
@alex-snezhko What you're seeing is differences in uninitialized bytes that we get back from `malloc`. I addressed this in #2045.